src/components/sets/sets.js

```diff
@@ -586,13 +603,17 @@ export default class SetsTree {
   emitVisibilityUpdate() {
     if (this.onVisibilityChange) {
       let cellIds = [];
+      const cellColors = {};
       this.visibleKeys.forEach((setKey) => {
         const node = this.findNode(setKey);
         if (node && node.set && node.set.length > 0) {
           cellIds = [...cellIds, ...node.set];
+          node.set.forEach((cellId) => {
+            cellColors[cellId] = node.color;
+          });
         }
       });
-      this.onVisibilityChange(new Set(cellIds));
+      this.onVisibilityChange(new Set(cellIds), cellColors);
```

Comment:
... or better yet, use `fromEntries`, so it's clear that we're just constructing a data structure, and the loop per-se is not the point.